spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseRewriteDataFilesSparkAction.java

```diff
@@ -352,6 +354,27 @@ private Result doExecuteWithPartialProgress(RewriteExecutionContext ctx, Stream<
             return new RewriteFileGroup(info, tasks);
           });
         });
+
+    return rewriteFileGroupStream.sorted(rewriteGroupComparator());
+  }
+
+  private Comparator<RewriteFileGroup> rewriteGroupComparator() {
+    if (rewriteJobOrder.equals(RewriteJobOrder.BYTES_ASC.orderName())) {
+      // Rewrite the smallest job groups first
```

Comment:
docs probably aren't needed here, I think everything will be clear when we change the RewriteFileGroup method names